pkg/controller/common/tracing/transaction.go

```diff
@@ -28,3 +28,26 @@ func EndTransaction(tx *apm.Transaction) {
 		tx.End()
 	}
 }
+
+// NewContextTransaction starts a new transaction and sets up a new context with that transaction that also contains the related
+// APM agent's tracer.
+func NewContextTransaction(t *apm.Tracer, txType, txName string, labels map[string]string) context.Context {
```

Comment:
I wanted to keep them both until we have the improvements (setting label) that were added to Agent controller elsewhere. What I want to avoid is to bundle the rework of other controllers into this PR. Similar to `LogReconciliationRunNoSideEffects` function.

Reply:
OK makes sense!